### PR TITLE
paperless-ngx: update to 1.8.0, python3Packages.django-q: add gador as maintainer, switch to GitHub

### PR DESCRIPTION
**Pull request by gador**
###### Description of changes

Update paperless-ngx [Changelog](https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.8.0). I had to switch to `fetchFromGitHub` for django-q, so the override in paperless-ngx works.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by erikarvstedt — APPROVED** [review]

File: pkgs/development/python-modules/django-q/default.nix [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
-    django-picklefield arrow blessed django future
+    django-picklefield
+    arrow
+    blessed
+    django
+    future
   ];
 
+  # all checks need a running redis instance
   doCheck = false;
```
Comment (risicle) [inline pkgs/development/python-modules/django-q/default.nix:35]: Adding a `pythonImportsCheck` would be better than nothing.

File: pkgs/applications/office/paperless-ngx/default.nix [hunk 2]
```diff
@@ -19,6 +20,17 @@ let
     packageOverrides = self: super: {
       django = super.django_4;
 
+      # use paperless-ngx version of django-q
+      # see https://github.com/paperless-ngx/paperless-ngx/pull/1014
+      django-q = super.django-q.overridePythonAttrs (oldAttrs: rec {
+        src = fetchFromGitHub {
```
Comment (risicle) [inline pkgs/applications/office/paperless-ngx/default.nix:26]: Should maybe also change the `version`? Not sure what to...

**Comment by risicle** [comment]
This should be two commits - first the `django-q` one, then the `paperless-ngx` one (for which the commit message should be `paperless-ngx: 1.7.1 -> 1.8.0`)

File: pkgs/development/python-modules/django-q/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/development/python-modules/django-q/default.nix:35]: Also FWIW, we actually _can_ run a redis server for the `checkPhase`, see how `django-redis` does it.

**Comment by risicle** [comment]
On nixos x86_64:

```
builder for '/nix/store/08vaqjyaqwdrhb1xc8vjngxzkdd5di4y-python3.10-HyperKitty-1.3.5.drv' failed with exit code 1; last 10 log lines:
  Requirement already satisfied: robot-detection>=0.3 in /nix/store/1zz2jpn01s2prbqnh3dxqakimradw54z-python3.10-robot-detection-0.4/lib/python3.10/site-packages (from HyperKitty==1.3.5) (0.4)
  Requirement already satisfied: django-compressor>=1.3 in /nix/store/nk954l1mqlqrsvshnj4rxbjc6hrqk85z-python3.10-django-compressor-4.1/lib/python3.10/site-packages (from HyperKitty==1.3.5) (4.1)
  Requirement already satisfied: networkx>=2.0 in /nix/store/m15pg6i8611sp68m4blsgr2bhf8yrb9s-python3.10-networkx-2.8.4/lib/python3.10/site-packages (from HyperKitty==1.3.5) (2.8.4)
  Requirement already satisfied: django-extensions>=1.3.7 in /nix/store/gck2sw46nx5ym1vy57ryi09qfcsa21fm-python3.10-django-extensions-3.2.0/lib/python3.10/site-packages (from HyperKitty==1.3.5) (3.2.0)
  Requirement already satisfied: pytz>=2012 in /nix/store/vzq0cn3r7zfcgcb8lc6gsg3sblrlnphz-python3.10-pytz-2022.1/lib/python3.10/site-packages (from HyperKitty==1.3.5) (2022.1)
  Requirement already satisfied: django-gravatar2>=1.0.6 in /nix/store/y9xg8408j2sq11a3w8mmqw5awrq655qa-python3.10-django-gravatar2-1.4.4/lib/python3.10/site-packages (from HyperKitty==1.3.5) (1.4.4)
  Requirement already satisfied: mistune>=2.0.0rc1 in /nix/store/7980wp7s2mdpddk3ns15iwsfp3c097l5-python3.10-mistune-2.0.4/lib/python3.10/site-packages (from HyperKitty==1.3.5) (2.0.4)
  ERROR: Could not find a version that satisfies the requirement django-q>=1.0.0 (from hyperkitty) (from versions: none)
  ERROR: No matching distribution found for django-q>=1.0.0
  

```

This causes `mailman-web` to fail to build.

File: pkgs/development/python-modules/django-q/default.nix (hunk 1 above)
Comment (gador) [reply pkgs/development/python-modules/django-q/default.nix]: good idea, thanks! I added the importCheck and the redis server to the checks

File: pkgs/applications/office/paperless-ngx/default.nix (hunk 2 above)
Comment (gador) [reply pkgs/applications/office/paperless-ngx/default.nix:26]: I would leave it, as is. The fork didn't change the version either, and paperless-ngx is a leaflet package, so it shouldn't conflict with any downstream packages

**Comment by gador** [comment]
> This should be two commits - first the `django-q` one, then the `paperless-ngx` one (for which the commit message should be `paperless-ngx: 1.7.1 -> 1.8.0`)

Done

**Comment by gador** [comment]
> On nixos x86_64:
> 
> ```
> builder for '/nix/store/08vaqjyaqwdrhb1xc8vjngxzkdd5di4y-python3.10-HyperKitty-1.3.5.drv' failed with exit code 1; last 10 log lines:
>   Requirement already satisfied: robot-detection>=0.3 in /nix/store/1zz2jpn01s2prbqnh3dxqakimradw54z-python3.10-robot-detection-0.4/lib/python3.10/site-packages (from HyperKitty==1.3.5) (0.4)
>   Requirement already satisfied: django-compressor>=1.3 in /nix/store/nk954l1mqlqrsvshnj4rxbjc6hrqk85z-python3.10-django-compressor-4.1/lib/python3.10/site-packages (from HyperKitty==1.3.5) (4.1)
>   Requirement already satisfied: networkx>=2.0 in /nix/store/m15pg6i8611sp68m4blsgr2bhf8yrb9s-python3.10-networkx-2.8.4/lib/python3.10/site-packages (from HyperKitty==1.3.5) (2.8.4)
>   Requirement already satisfied: django-extensions>=1.3.7 in /nix/store/gck2sw46nx5ym1vy57ryi09qfcsa21fm-python3.10-django-extensions-3.2.0/lib/python3.10/site-packages (from HyperKitty==1.3.5) (3.2.0)
>   Requirement already satisfied: pytz>=2012 in /nix/store/vzq0cn3r7zfcgcb8lc6gsg3sblrlnphz-python3.10-pytz-2022.1/lib/python3.10/site-packages (from HyperKitty==1.3.5) (2022.1)
>   Requirement already satisfied: django-gravatar2>=1.0.6 in /nix/store/y9xg8408j2sq11a3w8mmqw5awrq655qa-python3.10-django-gravatar2-1.4.4/lib/python3.10/site-packages (from HyperKitty==1.3.5) (1.4.4)
>   Requirement already satisfied: mistune>=2.0.0rc1 in /nix/store/7980wp7s2mdpddk3ns15iwsfp3c097l5-python3.10-mistune-2.0.4/lib/python3.10/site-packages (from HyperKitty==1.3.5) (2.0.4)
>   ERROR: Could not find a version that satisfies the requirement django-q>=1.0.0 (from hyperkitty) (from versions: none)
>   ERROR: No matching distribution found for django-q>=1.0.0
>   
> ```
> 
> This causes `mailman-web` to fail to build.

Fixed. The error came from a wrongful interpretation of the `pyproject.toml` file.

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 185669` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>mailman-web</li>
<li>paperless-ngx</li>
<li>python310Packages.django-q</li>
<li>python39Packages.django-q</li>
</ul>
</details>

**Comment by risicle** [comment]
Think we'll have to disable the `django-q` tests on darwin - they fail and also have the quirk that because `postCheck` doesn't run, the zombie redis is left running and the build never terminates. This isn't a problem on linux.

**Comment by gador** [comment]
@risicle done

**Review by erikarvstedt — APPROVED** [review]

**Review by risicle — APPROVED** [review]
`nixpkgs-review` reveals no new failures, macos 10.15 & nixos x86_64

**Pull request merged**